0.139.0 -  "panic: bug: unexpectedly failed to get releases for id"

Once we upgraded, all our pipelines started failing.

Version: 0.138.7

```
❯ helmfile -e qa3 --selector name=mauka status
Getting status mauka
```

Version: 0.139.0

```
❯ helmfile -e qa3 --selector name=mauka status
panic: bug: unexpectedly failed to get releases for id "kube-system/high-priority": [qa3-mauka/mauka]

goroutine 1 [running]:
github.com/roboll/helmfile/pkg/state.GroupReleasesByDependency(0xc0000b9800, 0x1, 0x1, 0x10000, 0x0, 0x0, 0x0, 0x1, 0x1, 0x0, ...)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/state/state_run.go:248 +0x1ac5
github.com/roboll/helmfile/pkg/state.SortedReleaseGroups(0xc0000b9800, 0x1, 0x1, 0x10000, 0x0, 0x0, 0x0, 0xc0005e52c0, 0x20, 0x37b5010, ...)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/state/state_run.go:126 +0x91
github.com/roboll/helmfile/pkg/state.(*HelmState).PlanReleases(0xc00039d8c0, 0x10000, 0x0, 0x0, 0x0, 0x100fa38, 0x10, 0x20bbee0, 0xc00015a601, 0xc0005e8530)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/state/state_run.go:115 +0xa9
github.com/roboll/helmfile/pkg/app.withDAG(0xc00039d8c0, 0x26fcc88, 0xc00053e840, 0xc000011168, 0x10000, 0x0, 0x0, 0x0, 0xc0005e8530, 0x0, ...)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:872 +0x7d
github.com/roboll/helmfile/pkg/app.(*App).status(0xc00021e480, 0xc0001e2200, 0x26d86d0, 0xc0002b6ea0, 0xc00015ac08, 0xc0002c66c8, 0xadec0e9b42460901, 0xad0000c00015aa58)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:1540 +0x4b1
github.com/roboll/helmfile/pkg/app.(*App).Status.func1.1()
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:385 +0x5b
github.com/roboll/helmfile/pkg/app.(*Run).withPreparedCharts(0xc0001e2200, 0x224974f, 0x6, 0x10100, 0x0, 0x0, 0x0, 0x0, 0xc00015acc8, 0x0, ...)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/run.go:91 +0x43d
github.com/roboll/helmfile/pkg/app.(*App).Status.func1(0xc0001e2200, 0xc0001e2200, 0x0, 0x0, 0x0)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:381 +0x105
github.com/roboll/helmfile/pkg/app.(*App).ForEachState.func1(0xc00039d8c0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:843 +0xda
github.com/roboll/helmfile/pkg/app.(*App).visitStatesWithSelectorsAndRemoteSupport.func1.1(0xc00039d8c0, 0xc00053e840, 0xc00039d8c0, 0xc0002cef00)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:956 +0x2f
github.com/roboll/helmfile/pkg/app.processFilteredReleases(0xc00039d8c0, 0x26fcc88, 0xc00053e840, 0xc00015ae60, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:977 +0x128
github.com/roboll/helmfile/pkg/app.(*App).visitStatesWithSelectorsAndRemoteSupport.func1(0xc00039d8c0, 0xc00039d8c0, 0x0, 0x0, 0xc000352840)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:955 +0x8f
github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1(0x225185a, 0xd, 0xc0001aa1b0, 0x2e, 0x0, 0x0)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:791 +0x82d
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles.func1(0x22456ae, 0x1)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:623 +0xa2
github.com/roboll/helmfile/pkg/app.(*App).within(0xc00021e480, 0x22456ae, 0x1, 0xc00055d4d8, 0xc00055d498, 0x2)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:565 +0x629
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles(0xc00021e480, 0x0, 0x0, 0xc0002b6ce0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:617 +0x2ee
github.com/roboll/helmfile/pkg/app.(*App).visitStates(0xc00021e480, 0x0, 0x0, 0xc0002b6ce0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:703 +0x125
github.com/roboll/helmfile/pkg/app.(*App).visitStatesWithSelectorsAndRemoteSupport(0xc00021e480, 0x0, 0x0, 0xc00055d7f8, 0xc00055d868, 0x1, 0x1, 0x10, 0x20bbf60)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:962 +0x4fd
github.com/roboll/helmfile/pkg/app.(*App).ForEachState(0xc00021e480, 0xc00055d870, 0xc00055d868, 0x1, 0x1, 0x100cdec, 0x2231d00)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:839 +0xd6
github.com/roboll/helmfile/pkg/app.(*App).Status(0xc00021e480, 0x26d86d0, 0xc0002b6ea0, 0xc0002b6ea0, 0xc000011168)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/pkg/app/app.go:380 +0xad
main.main.func11(0xc00021e480, 0xc0001269a0, 0x0, 0xc0002b6e50, 0x0)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/main.go:536 +0x6d
main.action.func1(0xc0001269a0, 0xc000501d00, 0xc000501d20)
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/main.go:928 +0xdc
github.com/urfave/cli.HandleAction(0x1fdd360, 0xc0002b66e0, 0xc0001269a0, 0xc0001269a0, 0x0)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0x224974f, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x227bd20, 0x29, 0x0, ...)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).Run(0xc000132700, 0xc0001ae120, 0x6, 0x6, 0x0, 0x0)
	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.main()
	/private/tmp/helmfile-20210506-26861-1vo93z9/helmfile-0.139.0/main.go:667 +0x50d7
```

```
❯ helm list -A --all-namespaces | grep qa3-mauka
mauka                         	qa3-mauka        	377     	2021-05-06 06:08:01.053141 -0400 EDT   	deployed	mauka-0.2.13                       	2.0
```

```
❯ helm list -A --all-namespaces | grep high-priority
high-priority                 	kube-system      	1       	2020-10-12 12:50:33.147348 -0400 EDT   	deployed	raw-0.2.3                          	0.2.3
```

--- a/pkg/app/app.go
+++ b/pkg/app/app.go
@@ -1447,13 +1447,12 @@ func (a *App) lint(r *Run, c LintConfigProvider) (bool, []error) {
 	// on running various helm commands on unnecessary releases
 	st.Releases = selectedReleases
 
-	releasesToRender := map[string]state.ReleaseSpec{}
+	var toLint []state.ReleaseSpec
 	for _, r := range selectedReleases {
-		id := state.ReleaseToID(&r)
 		if r.Installed != nil && !*r.Installed {
 			continue
 		}
-		releasesToRender[id] = r
+		toLint = append(toLint, r)
 	}
 
 	var errs []error
@@ -1470,18 +1469,8 @@ func (a *App) lint(r *Run, c LintConfigProvider) (bool, []error) {
 		helm.SetExtraArgs(args...)
 	}
 
-	if len(releasesToRender) > 0 {
-		_, templateErrs := withDAG(st, helm, a.Logger, state.PlanOptions{Reverse: false, SkipNeeds: true}, a.Wrap(func(subst *state.HelmState, helm helmexec.Interface) []error {
-			var rs []state.ReleaseSpec
-
-			for _, r := range subst.Releases {
-				if r2, ok := releasesToRender[state.ReleaseToID(&r)]; ok {
-					rs = append(rs, r2)
-				}
-			}
-
-			subst.Releases = rs
-
+	if len(toLint) > 0 {
+		_, templateErrs := withDAG(st, helm, a.Logger, state.PlanOptions{SelectedReleases: toLint, Reverse: false, SkipNeeds: true}, a.WrapWithoutSelector(func(subst *state.HelmState, helm helmexec.Interface) []error {
 			opts := &state.LintOpts{
 				Set: c.Set(),
 			}
@@ -1513,13 +1502,12 @@ func (a *App) status(r *Run, c StatusesConfigProvider) (bool, []error) {
 	// on running various helm commands on unnecessary releases
 	st.Releases = selectedAndNeededReleases
 
-	releasesToRender := map[string]state.ReleaseSpec{}
+	var toStatus []state.ReleaseSpec
 	for _, r := range selectedReleases {
-		id := state.ReleaseToID(&r)
 		if r.Installed != nil && !*r.Installed {
 			continue
 		}
-		releasesToRender[id] = r
+		toStatus = append(toStatus, r)
 	}
 
 	var errs []error
@@ -1536,18 +1524,8 @@ func (a *App) status(r *Run, c StatusesConfigProvider) (bool, []error) {
 		helm.SetExtraArgs(args...)
 	}
 
-	if len(releasesToRender) > 0 {
-		_, templateErrs := withDAG(st, helm, a.Logger, state.PlanOptions{Reverse: false, SkipNeeds: true}, a.Wrap(func(subst *state.HelmState, helm helmexec.Interface) []error {
-			var rs []state.ReleaseSpec
-
-			for _, r := range subst.Releases {
-				if r2, ok := releasesToRender[state.ReleaseToID(&r)]; ok {
-					rs = append(rs, r2)
-				}
-			}
-
-			subst.Releases = rs
-
+	if len(toStatus) > 0 {
+		_, templateErrs := withDAG(st, helm, a.Logger, state.PlanOptions{SelectedReleases: toStatus, Reverse: false, SkipNeeds: true}, a.WrapWithoutSelector(func(subst *state.HelmState, helm helmexec.Interface) []error {
 			return subst.ReleaseStatuses(helm, c.Concurrency())
 		}))
 

